### ISSUE TITLE
Input method ignored

### ISSUE THREAD

**Issue by thierry-FreeBSD** [mike-fabian/ibus-typing-booster#41]
Since release 2.3.1 (same with 2.3.3), even with fr-azerty (AZ) as the selected input method, the keyboard is in QWERTY mode.
It is OK when NoIme is selected (my native keyboard is AZERTY).

**Comment by mike-fabian** [comment]
That is very surprising to me, I can’t imagine anything I changed could do this.
Trying to reproduce this now ...

**Comment by mike-fabian** [comment]
This seems to be normal behaviour.

As your native keyboard is already AZERTY, you don’t need the
fr-azerty and should not use it. fr-azerty is only intended to be used
with US English keyboard layouts. I.e. fr-azerty simulates a French
AZERTY layout on top of an US English layout.

You can see that when looking at the help for fr-azerty.

Select fr-azerty in the setup of ibus-typing-booster and click on the
“Input method help” button (“Aide pour la méthode de saisie”).

In the help window you get you will find:

   (description (_"Simulating Azerty keyboard on English keyboard.


(map
 (normal
  
  ("q" ?a)

  ("a" ?q)

  ("Q" ?A)

  ("A" ?Q)

So this input method replaces a 'q' coming as input with an 'a' and
an 'a' coming as input with a 'q'.

If the keyboard layout you are using is QWERTY,  this does  the right
thing to simulate an AZERTY layout.

*But* if you are already using an AZERTY layout, it makes no sense
at all, because the incoming 'a' when you type the key at the top
left of the AZERTY layout will be changed into a 'q', which
makes no sense at all.

This "fr-azerty" input method only makes sense when used on an English
layout, I am not even sure whether it works on all English
layouts. There may be subtle differences between the US English layout
and the GB English layout which cause problems with "fr-azerty", I
think "fr-azerty" is really only designed to be used with an US
English layout.

So in short, if already have an AZERTY layout, just don‘t use
"fr-azerty", then there is no problem.

And this is not really a bug.

**Comment by mike-fabian** [comment]
Closing, because it works  as expected after all.

**Comment by thierry-FreeBSD** [comment]
Thanks for your explanation and sorry for the noise!
